openlibrary/macros/LoanStatus.html

```diff
@@ -47,15 +53,19 @@
   </form>
 
 $elif ocaid and not page.is_access_restricted() and editions_page:
-  $ viewbook = "//%s/stream/XXX?ref=ol" % bookreader_host()
-  <a href="$viewbook.replace('XXX', ocaid)" title="Use BookReader to read online" class="cta-btn cta-btn--available">Read</a>
+  $ stream_url = "//%s/stream/%s?ref=ol" % (bookreader_host(), ocaid)
+  <div class="cta-button-group">
+    <a href="$stream_url" title="Use BookReader to read online" class="cta-btn cta-btn--available">Read</a>
+    $:listen_button(stream_url + "&", "Listen to this book using text-to-speech")
+  </div>
+
 
 
 $elif (availability_status == 'borrow_available') or my_turn_to_borrow:
-  <a href="$borrow_link" title="Borrow from $contributor" id="borrow_ebook"
-     class="cta-btn cta-btn--available">
-    Borrow
-  </a>
+  <div class="cta-button-group">
+    <a href="$borrow_link" title="Borrow from $contributor" id="borrow_ebook" class="cta-btn cta-btn--available">Borrow</a>
+    $:listen_button(borrow_link + "?", "Borrow this book and listen to it using text-to-speech")
```

Comment:
Does borrow_url not have `?ref=ol`? If not... it probably should?

It looks like maybe not because this is being tracked within `plugins/upstream/borrow`

Reply:
It wasn't before, so I'm going to leave as-is.